### ISSUE TITLE
Ensure graceful shutdown of `lnd` before electron quits

### ISSUE THREAD

**Issue by valentinewallace** [lightninglabs/lightning-app#802]
Only quit electron when we've successfully received the `exit` signal from the lnd child process.

**Comment by tanx** [comment]
Closing due to inactivity

**Comment by cfromknecht** [comment]
@tanx can we reopen this issue? the app currently doesn't complete a graceful shutdown of lnd when exiting.

this was partially addressed by #550, which properly sends the SIGINT. however, when closing the app, the logs only show: 
```
14423 2019-03-15 15:22:00.206 [INF] LTND: Received SIGINT (Ctrl+C).
14424 2019-03-15 15:22:00.206 [INF] LTND: Shutting down...
14425 2019-03-15 15:22:00.206 [INF] LTND: Gracefully shutting down.
```

So lnd received the request to shutdown, but since no other logs are found to me this indicates that the app isn't waiting for the child process to terminate on its own.

A graceful shutdown should result in the last line of the logs showing: `Shutdown complete`, like in this execution when I Ctrl-C the daemon directly: https://paste.ubuntu.com/p/7k4KBm4B8k/

**Comment by tanx** [comment]
@cfromknecht good catch! Yeah I've noticed that the lnd process keeps running in certain cases even when the app is closed.
